internal/sdk/cloudian/datasize.go

```diff
@@ -0,0 +1,44 @@
+package cloudian
+
+import (
+	"fmt"
+)
+
+// Construct by e.g. 3 * TB
+type ByteSize uint64
```

Comment:
I don't think we need this custom type in the SDK. To provide better usability, we could use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/intstr#IntOrString in the provider API (Crossplane).